orchagent/fdborch.cpp

```diff
@@ -139,132 +179,237 @@ void FdbOrch::update(sai_fdb_event_t        type,
         }
 
         update.add = true;
-        update.entry.port_name = update.port.m_alias;
-        storeFdbEntryState(update);
+        update.type = "dynamic";
+        update.port.m_fdb_count++;
+        m_portsOrch->setPort(update.port.m_alias, update.port);
+        vlan.m_fdb_count++;
+        m_portsOrch->setPort(vlan.m_alias, vlan);
 
-        SWSS_LOG_INFO("Notifying observers of FDB entry LEARN");
-        for (auto observer: m_observers)
-        {
-            observer->update(SUBJECT_TYPE_FDB_CHANGE, &update);
-        }
+        storeFdbEntryState(update);
+        notify(SUBJECT_TYPE_FDB_CHANGE, &update);
 
         break;
-
+    }
     case SAI_FDB_EVENT_AGED:
-    case SAI_FDB_EVENT_MOVE:
-        update.add = false;
-        storeFdbEntryState(update);
+    {
+        SWSS_LOG_INFO("Received AGE event for bvid=%lx mac=%s port=%lx", entry->bv_id, update.entry.mac.to_string().c_str(), bridge_port_id);
 
-        SWSS_LOG_INFO("Notifying observers of FDB entry removal on AGED/MOVED");
-        for (auto observer: m_observers)
+        if (!m_portsOrch->getPort(entry->bv_id, vlan))
         {
-            observer->update(SUBJECT_TYPE_FDB_CHANGE, &update);
+            SWSS_LOG_NOTICE("FdbOrch AGE notification: Failed to locate vlan port from bv_id 0x%lx", entry->bv_id);
         }
 
-        break;
+        auto existing_entry = m_entries.find(update.entry);
+        // we don't have such entries
+        if (existing_entry == m_entries.end())
+        {
+             SWSS_LOG_INFO("FdbOrch AGE notification: mac %s is not present in bv_id 0x%lx bp 0x%lx",
+                    update.entry.mac.to_string().c_str(), entry->bv_id, bridge_port_id);
+             break;
+        }
 
-    case SAI_FDB_EVENT_FLUSHED:
+        if (!m_portsOrch->getPortByBridgePortId(bridge_port_id, update.port))
+        {
+            SWSS_LOG_NOTICE("FdbOrch AGE notification: Failed to get port by bridge port ID 0x%lx", bridge_port_id);
+        }
 
-        SWSS_LOG_INFO("FDB Flush event received: [ %s , 0x%" PRIx64 " ], \
-                       bridge port ID: 0x%" PRIx64 ".",
-                       update.entry.mac.to_string().c_str(), entry->bv_id,
-                       bridge_port_id);
+        if (existing_entry->second.bridge_port_id != bridge_port_id)
+        {
+            SWSS_LOG_NOTICE("FdbOrch AGE notification: Stale aging event received for mac-bv_id %s-0x%lx with bp=0x%lx existing bp=0x%lx", update.entry.mac.to_string().c_str(), entry->bv_id, bridge_port_id, existing_entry->second.bridge_port_id);
+            // We need to get the port for bridge-port in existing fdb
+            if (!m_portsOrch->getPortByBridgePortId(existing_entry->second.bridge_port_id, update.port))
+            {
+                SWSS_LOG_NOTICE("FdbOrch AGE notification: Failed to get port by bridge port ID 0x%lx", existing_entry->second.bridge_port_id);
+            }
+            // dont return, let it delete just to bring SONiC and SAI in sync
```

Comment:
Delete this commented out code? I wont mind keeping it as it helps the reader!

Reply:
I prefer to keep it; hence kept.